### ISSUE TITLE
Fix false positives for nested literals in unit-no-unknown

### ISSUE THREAD

**Issue by TLadd** [stylelint/stylelint#4728]
> Clearly describe the bug

I'm getting a `unit-no-unknown` lint error when using string interpolation to calculate an animation delay.

> Which rule, if any, is the bug related to?

`unit-no-unknown`

> What code is needed to reproduce the bug?

```js
import * as styled from "styled-components";

const StyledDiv = styled.div`
  animation-delay: ${({ index }) => `${index * 0.5}s`};
`;
```

I have a repository illustrating the issue here: https://github.com/TLadd/stylelint-no-unknown-bug. Can clone the repo and run
```shell
yarn
yarn lint:css
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-recommended",
    "stylelint-config-styled-components"
  ]
}
```

> Which version of stylelint are you using?

13.3.3

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

stylelint './src/**/*.js'

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Css-in-JS

> What did you expect to happen?

Not to receive a lint error

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
test.css
✖  Unexpected unknown unit "s`"   unit-no-unknown
```

**Comment by jeddy3** [comment]
@TLadd Thanks for the report and for making a reproducible repo.

I can also reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICWBbADhATgFwAQCpcBDAZ1xOwE8AbGKXAM0whVwB0QKa6BaSdCADsYg7CQ4BuNoOmRBFXAGUqtKABEkAN1wBeciroA6KFoAG03MUGoi2JEJ6xqRSvFwASYAApguJINgAD1wAXwBKXQA+XFNPfyCCXAAGQwBWEJJTEKlBUwkQABpOSlEiQIRwEhIefx4AKxJC8CEGJABzCuALdhBMAFdacQRcLsFLSw4+62weQQgeKYBrOYB3GWHsfphukOkQkBCgA).

It's likely a problem with [the parser](https://github.com/stylelint/postcss-css-in-js) stylelint uses [to support CSS-in-JS](https://stylelint.io/about/syntaxes). I've added this issue to https://github.com/stylelint/stylelint/issues/4574, which is a growing list of the problems with the syntax that we need help to fix.

Please consider contributing to the parser if you have time.

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

We'll also [be deprecating](https://github.com/stylelint/postcss-css-in-js/issues/225) our [forked CSS-in-JS package](https://github.com/stylelint/postcss-css-in-js), in favour of leaner custom syntaxes.

Please consider [writing a custom syntax](https://github.com/stylelint/stylelint/blob/main/docs/developer-guide/syntaxes.md) specific to the CSS-in-JS library in this issue and fixing the bug there.
